pkg/eventingtls/trust_bundle.go

```diff
@@ -138,16 +140,20 @@ func PropagateTrustBundles(ctx context.Context, k8s kubernetes.Interface, trustB
 	return nil
 }
 
-func AddTrustBundleVolumes(trustBundleLister corev1listers.ConfigMapLister, obj kmeta.Accessor, pt *corev1.PodTemplateSpec) (*corev1.PodTemplateSpec, error) {
+func AddTrustBundleVolumes(trustBundleLister corev1listers.ConfigMapLister, obj kmeta.Accessor, pt *corev1.PodSpec) (*corev1.PodSpec, error) {
 	cms, err := trustBundleLister.ConfigMaps(obj.GetNamespace()).List(TrustBundleSelector)
 	if err != nil {
 		return nil, fmt.Errorf("failed to list trust bundles ConfigMaps in %q: %w", obj.GetNamespace(), err)
 	}
 
 	pt = pt.DeepCopy()
 	for _, cm := range cms {
-		pt.Spec.Volumes = append(pt.Spec.Volumes, corev1.Volume{
-			Name: cm.Name,
+		volumeName := fmt.Sprintf("%s%s", TrustBundleVolumeNamePrefix, cm.Name)
+		if len(volumeName) > 63 { // 63 is the maximum name length (there isn't a constant for it)
+			volumeName = kmeta.ChildName(TrustBundleVolumeNamePrefix, cm.Name)
+		}
```

Comment:
nit: Any reason not using `volumeName := kmeta.ChildName(TrustBundleVolumeNamePrefix, cm.Name)` directly here, as `ChildName` only hashes when the name is too long?

Reply:
Done here https://github.com/knative/eventing/pull/7567